Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs

```diff
@@ -0,0 +1,230 @@
+            }
+            else
+            {
+                UpdateAppearance(uid, null, component);
+                _popupSystem.PopupEntity(component.MeatSource0, uid, Filter.Entities(user));
+            }
+
+            return true;
+        }
+
+        private void UpdateAppearance(EntityUid uid, AppearanceComponent? appearance = null, KitchenSpikeComponent? component = null)
+        {
+            if (!Resolve(uid, ref component, ref appearance, false))
+                return;
+            
+            appearance.SetData(KitchenSpikeVisuals.Status, (component.MeatParts > 0) ? KitchenSpikeStatus.Bloody : KitchenSpikeStatus.Empty);
+        }
+
+        private bool Spikeable(EntityUid uid, EntityUid userUid, EntityUid victimUid,
+            KitchenSpikeComponent? component = null, SharedButcherableComponent? butcherable = null)
+        {
+            if (!Resolve(uid, ref component))
+                return false;
+
+            if (component.MeatParts > 0)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-deny-collect", ("this", uid)), uid, Filter.Entities(userUid));
+                return false;
+            }
+
+            if (!Resolve(victimUid, ref butcherable, false) || butcherable.MeatPrototype == null)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-deny-butcher", ("victim", victimUid), ("this", uid)), victimUid, Filter.Entities(userUid));
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool TrySpike(EntityUid uid, EntityUid userUid, EntityUid victimUid, KitchenSpikeComponent? component = null,
+            SharedButcherableComponent? butcherable = null, MobStateComponent? mobState = null)
+        {
+            if (!Resolve(uid, ref component) || component.InUse ||
+                !Resolve(victimUid, ref butcherable) || butcherable.BeingButchered)
+                return false;
+
+            // THE WHAT? (again)
+            // Prevent dead from being spiked TODO: Maybe remove when rounds can be played and DOT is implemented
+            if (Resolve(victimUid, ref mobState, false) &&
+                !mobState.IsDead())
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-deny-not-dead", ("victim", victimUid)),
+                    victimUid, Filter.Entities(userUid));
+                return false;
+            }
+
+            if (userUid != victimUid)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-begin-hook-victim", ("user", userUid), ("this", uid)), victimUid, Filter.Entities(victimUid));
+            }
+            else
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-begin-hook-self", ("this", uid)), victimUid, Filter.Pvs(uid)); // This is actually unreachable and should be in SuicideEvent
```

Comment:
as this is reachable, it should just be deleted?

Reply:
I've commented it out for future